File: backend/app/models/room.py
from sqlalchemy import Column, Integer, String, Enum
from app.database import Base

class Room(Base):
    __tablename__ = "rooms"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(50), unique=True, nullable=False)
    capacity = Column(Integer)
    type = Column(Enum("lecture", "lab"), nullable=False)
    building = Column(String(100))

File: SmartTimetable/src/SmartTimetable.Core/Entities/Room.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartTimetable.Core.Entities;

[Table("rooms")]
public class Room
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [MaxLength(50)]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("capacity")]
    public int? Capacity { get; set; }

    [Required]
    [MaxLength(10)]
    [Column("type")]
    public string Type { get; set; } = "lecture";

    [MaxLength(100)]
    [Column("building")]
    public string? Building { get; set; }
}
